Formula/step.rb

```diff
@@ -3,6 +3,7 @@ class Step < Formula
   homepage "https://smallstep.com"
   url "https://github.com/smallstep/cli/archive/v0.8.5.tar.gz"
   sha256 "b809c4638ffd6d0c3e1bff66d87d9fbea186783735544492c34f35fe1e9f9d79"
+  head "https://github.com/smallstep/cli.git"
```

Comment:
@dopey: I'm curious what your reasoning for adding a head url is. Unless it makes sense to do so to improve software quality, we avoid adding heads to formula. If you plan to use it for improving software quality, e.g., in nightly testing to test for installation regressions and issues caused by build dependencies, then I'm all for adding it. However, unless there is a compelling reason that users might want/need head installations, I would prefer that it is removed. In the future we'll be adding very prominent warnings to HEAD installs saying that they are totally unsupported and you're on your own if something breaks.

Reply:
I actually grabbed that from another Formula because I thought it might be useful. Should have looked up what it was doing. Removed.

In the same vein I saw another formula using this url definition (https://github.com/Homebrew/homebrew-core/blob/master/Formula/goofys.rb#L6-L8) and was wondering if it wouldn't be nicer to switch to that over the tarball. Any drawbacks?

Reply:
> I actually grabbed that from another Formula because I thought it might be useful. Should have looked up what it was doing. Removed.

I like to add add nightly tests to my projects that build using `brew install --HEAD <my-package>` to ensure that upstream dependencies, or changes to homebrew didn't break the install procedure for the package in question. I support and encourage this approach, because then the package authors/maintainers catch potential breakage of their package when installed via Homebrew. This way, when they mint a new release, they will have been testing the homebrew installation all along.

However, I discourage adding in simply because "some user may want this". If users want development software, they should be comfortable enough with git to grab it for themselves. The exception to this, in my mind, is tooling stuff. I like knowing if stuff, e.g., in nightly GFortran, might break my projects so that I can update them ahead of the next major compiler release.

Reply:
Makes sense for that use case, and definitely not one I had in mind when I added it. Thanks for the explanation.

Reply:
> In the same vein I saw another formula using this url definition ([/Formula/goofys.rb@`master`#L6-L8](https://github.com/Homebrew/homebrew-core/blob/master/Formula/goofys.rb#L6-L8)) and was wondering if it wouldn't be nicer to switch to that over the tarball. Any drawbacks?

The way you have it now is probably more canonical. I don't recall whether or not `brew` will do a shallow clone using the tag and hash version scheme. If it doesn't, then grabbing a whole git repo will be slower and more time consuming than just the tarball, and it will take up more disk space. Unless there is a compelling reason to switch, let's leave the version & download as is.